spinnman/transceiver.py

```diff
@@ -2859,12 +2859,12 @@ def register_udp_listener(self, callback, connection_class,
         if connection is None:
             connection = connection_class(local_host=local_host)
             self._all_connections.add(connection)
-        listener = ConnectionListener(connection)
-        listener.start()
-        self._udp_receive_connections_by_port[connection.local_port][
-            local_host] = (connection, listener)
-        connections_of_class.append((connection, listener))
-        listener.add_callback(callback)
+            listener = ConnectionListener(connection)
+            listener.start()
+            self._udp_receive_connections_by_port[connection.local_port][
+                local_host] = (connection, listener)
+            connections_of_class.append((connection, listener))
+            listener.add_callback(callback)
```

Comment:
Testing on the same simulation, everything still ends gracefully with this unindented.

Therefore, provided it's logical to have multiple callbacks on the same listener, it seems sensible to have this line unindented - it may also explain why this piece of code is called multiple times.